pkg/reconciler/contour/config/contour.go

```diff
@@ -70,6 +83,37 @@ func NewContourFromConfigMap(configMap *corev1.ConfigMap) (*Contour, error) {
 		return nil, err
 	}
 
+	cors, ok := configMap.Data[corsPolicy]
+	if ok {
+		if err := yaml.Unmarshal([]byte(cors), &contourCORSPolicy); err != nil {
+			return nil, err
+		}
+
+		fields := [][]string{
+			contourCORSPolicy.AllowOrigin,
+			contourCORSPolicy.AllowMethods,
+			contourCORSPolicy.AllowHeaders,
+			contourCORSPolicy.ExposeHeaders,
+		}
+		for _, field := range fields {
+			if len(field) > 0 {
+				var validOption = regexp.MustCompile("^[a-zA-Z0-9!#$%&'*+.^_`|~-]+$")
+				for _, option := range field {
+					if !validOption.MatchString(option) {
+						return nil, fmt.Errorf("option %s is not validly formated", option)
```

Comment:
error messages here need to highlight that this is part of the `corsPolicy` attribute

Reply:
addressed